lib/pengine/pe_output.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2004-2018 Andrew Beekhof <andrew@beekhof.net>
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include <crm/pengine/internal.h>
+
+#define PE__VARIANT_BUNDLE 1
+#include "./variant.h"
```

Comment:
This macro is defined in the lib/pengine/bundle.c to import the `struct pe__bundle_variant_data_t`, I thought it would be fine to also use repeat it in the pe_output.c. If I cannot use the `struct pe__bundle_variant_data_t` pe_output.c, I will create a function back in boundle.c that will loops over the boundle_data and pass the pointer to this function as the last parameter. Maybe it will a little over-complicate the logic. Should I do this?

Reply:
You can move the entire function that needs it to bundle.c -- they don't all have to be in this file. There's a lib/pengine/pe_status_private.h file to declare functions that are only needed inside the library.

Reply:
I removed the output member of `resource_object_functions_s` and `pe__output_native, pe__output_native, pe__output_native, and pe__output_native`. Now the messages are called instead. For example
```
out->message(out, crm_element_name(replica->ip->xml), options, replica->ip);
out->message(out, crm_element_name(replica->child->xml), options, replica->child);
out->message(out, crm_element_name(replica->container->xml), options, replica->container);
out->message(out, crm_element_name(replica->remote->xml), options, replica->remote);
```

Reply:
I moved the whole function to the bundle.c